app/src/main/java/com/spascoding/taskycourse/feature_auth/domain/util/UserDataValidator.kt

```diff
@@ -20,7 +20,7 @@ object UserDataValidator {
     }
 
     fun validateEmail(text: String): Result<Unit, EmailError> {
-        if (!Patterns.EMAIL_ADDRESS.matcher(text).matches()) {
+        if (!PatternsCompat.EMAIL_ADDRESS.matcher(text).matches()) {
```

Comment:
Strictly speaking, Android dependencies still don't belong in domain, but if this `PatternsCompat` class fixes the issue with the local unit tests, I guess that's a fair trade off